cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/test/java/co/cask/cdap/security/authorization/sentry/binding/SentryAuthorizerTest.java

```diff
@@ -272,6 +278,46 @@ public void testHierarchy() throws Exception {
                      getUser("readers1"), Action.READ);
   }
 
+  @Test
+  public void testCache() throws Exception {
+    final NamespaceId ns1 = new NamespaceId("ns1");
+    final Principal alice = getUser("alice");
+    final Principal admin1 = getUser("admin1");
+    try {
+      assertUnauthorized(ns1, alice, Action.READ);
+      assertAuthorized(ns1, admin1, Action.ADMIN);
+    } catch (UnauthorizedException ignored) {
+      //
+    }
+    Assert.assertTrue(2 == authorizer.cacheAsMap().size());
+    Assert.assertTrue(authorizer.cacheAsMap().containsKey(new SentryAuthorizer.AuthorizationPrivilege(alice, ns1,
+                                                                                                       Action.READ)));
+    Assert.assertTrue(authorizer.cacheAsMap().containsKey(new SentryAuthorizer.AuthorizationPrivilege(admin1, ns1,
+                                                                                                      Action.ADMIN)));
+    Assert.assertFalse(authorizer.cacheAsMap().containsKey(new SentryAuthorizer.AuthorizationPrivilege(admin1, ns1,
+                                                                                                      Action.READ)));
+    Assert.assertFalse(authorizer.cacheAsMap().containsKey(new SentryAuthorizer.AuthorizationPrivilege(admin1, ns1,
+                                                                                                       Action.WRITE)));
+    Assert.assertFalse(authorizer.cacheAsMap().containsKey(new SentryAuthorizer.AuthorizationPrivilege(
+      admin1, ns1, Action.EXECUTE)));
+    TimeUnit.SECONDS.sleep(CACHE_TTL_SECS);
+    Assert.assertFalse(authorizer.cacheAsMap().containsKey(new SentryAuthorizer.AuthorizationPrivilege(alice, ns1,
+                                                                                                       Action.READ)));
+    Assert.assertFalse(authorizer.cacheAsMap().containsKey(new SentryAuthorizer.AuthorizationPrivilege(admin1, ns1,
+                                                                                                       Action.ADMIN)));
+    try {
+      assertUnauthorized(ns1, admin1, Action.READ);
+    } catch (UnauthorizedException ignored) {
+      //
+    }
+    Assert.assertTrue(authorizer.cacheAsMap().containsKey(new SentryAuthorizer.AuthorizationPrivilege(admin1, ns1,
+                                                                                                       Action.READ)));
+    Assert.assertFalse(authorizer.cacheAsMap().containsKey(new SentryAuthorizer.AuthorizationPrivilege(admin1, ns1,
+                                                                                                      Action.ADMIN)));
+    Assert.assertFalse(authorizer.cacheAsMap().containsKey(new SentryAuthorizer.AuthorizationPrivilege(admin1, ns1,
+                                                                                                       Action.WRITE)));
```

Comment:
should test the cache update on grant too ? Although might not be possible since this follows the file based storage and grant can't be done correct ?